test/memory/MemoryTestPublisher.cpp

```diff
@@ -0,0 +1,393 @@
+#endif
+
+#define TIME_LIMIT_US 10000
+
+using namespace eprosima;
+using namespace eprosima::fastrtps;
+using namespace eprosima::fastrtps::rtps;
+
+
+MemoryTestPublisher::MemoryTestPublisher():
+    mp_participant(nullptr),
+    mp_datapub(nullptr),
+    mp_commandpub(nullptr),
+    mp_commandsub(nullptr),
+    mp_memory(nullptr),
+    n_subscribers(0),
+    n_samples(0),
+    disc_count_(0),
+    comm_count_(0),
+    data_count_(0),
+    m_status(0),
+    n_received(0),
+    m_datapublistener(nullptr),
+    m_commandpublistener(nullptr),
+    m_commandsublistener(nullptr),
+    m_data_size(0)
+{
+    m_datapublistener.mp_up = this;
+    m_commandpublistener.mp_up = this;
+    m_commandsublistener.mp_up = this;
+    m_exportPrefix = "";
+}
+
+MemoryTestPublisher::~MemoryTestPublisher()
+{
+    Domain::removeParticipant(mp_participant);
+}
+
+
+bool MemoryTestPublisher::init(int n_sub, int n_sam, bool reliable, uint32_t pid, bool hostname, bool export_csv,
+        const std::string& export_prefix, const PropertyPolicy& part_property_policy,
+        const PropertyPolicy& property_policy, const std::string& sXMLConfigFile, uint32_t data_size)
+{
+    m_sXMLConfigFile = sXMLConfigFile;
+    n_samples = n_sam;
+    n_subscribers = n_sub;
+    n_export_csv = export_csv;
+    m_exportPrefix = export_prefix;
+    reliable_ = reliable;
+    m_data_size = data_size;
+
+    if (m_sXMLConfigFile.length() > 0)
+    {
+        // Create RTPSParticipant
+        std::string participant_profile_name = "participant_profile";
+        mp_participant = Domain::createParticipant(participant_profile_name);
+        if (mp_participant == nullptr)
+        {
+            return false;
+        }
+
+        // Register the type
+        Domain::registerType(mp_participant, (TopicDataType*)&memory_t);
+        Domain::registerType(mp_participant, (TopicDataType*)&command_t);
+
+        // Create Sending Publisher
+        std::string profile_name = "publisher_profile";
+        mp_datapub = Domain::createPublisher(mp_participant, profile_name, (PublisherListener*)&this->m_datapublistener);
+        if (mp_datapub == nullptr)
+        {
+            return false;
+        }
+        std::cout << "Publisher created" << std::endl;
+
+        // Create Command Publisher
+        profile_name = "publisher_cmd_profile";
+        mp_commandpub = Domain::createPublisher(mp_participant, profile_name, (PublisherListener*)&this->m_commandpublistener);
+        if (mp_commandpub == nullptr)
+        {
+            return false;
+        }
+        std::cout << "Publisher created" << std::endl;
+
+        profile_name = "subscriber_cmd_profile";
+        mp_commandsub = Domain::createSubscriber(mp_participant, profile_name, &this->m_commandsublistener);
+        if (mp_commandsub == nullptr)
+        {
+            return false;
+        }
+    }
+    else
+    {
+        ParticipantAttributes PParam;
+        PParam.rtps.defaultSendPort = 10042;
```

Comment:
Minimize the configuration. Except domainId, properties and name, the rest can be removed.